doc/reference/usermode/syscalls.rst

```diff
@@ -292,14 +292,6 @@ Several macros exist to validate arguments:
   instance, to validate the GPIO driver, one could use the
   :c:macro:`Z_SYSCALL_DRIVER_GPIO()` macro.
 
-* :c:macro:`Z_SYSCALL_SPECIFIC_DRIVER()` is a runtime check to verify that
```

Comment:
As I noted earlier added in #12056 in response to my #11993 but I never got information on how it was intended to be used, e.g. examples/tests (@andrewboie).

We have a solution from #18186 that defines how to add device-specific API extensions, but it does not address supporting the extensions in user-mode.  I will have to update #17631 to implement system call checks correctly for the extended API.  I will be looking at this very soon, but the solution may use this macro.

Also this has been present in releases including 1.14, so it may be used in out of tree drivers and so arguably cannot be removed without deprecation.

Reply:
I can deprecate it but it will need to verify against api and not init function anymore then. (as it was before I removed it entirely)

Reply:
If this must be removed it'll be because it can't be made to work, but the problem it was intended to help with is still present.   Based on research summarized in #24539 I propose this PR just switch it to checking the API instead of the init pointer, and not deprecate it.

Reply:
Ok, back to API check then.